### DISCUSSION TITLE
Fail to change the remote URL.

### DISCUSSION THREAD

**Discussion by youyungui** [community/community#143146] [Repositories]
### Select Topic Area

General

### Body

Issue: I encountered a critical issue when using GitHub repository to sync Logseq, with the following error message: fatal: unable to access 'https://github.com.cnpmjs.org/youyungui/My-Knowledge-System.git/': LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to github.com.cnpmjs.org:443. It appears that the root of the problem is that the remote URL of the GitHub repository was changed, causing sync issues.

Environment: I am using the latest version of macOS, 15.1, configured via both the web and GitHub desktop client.

Issue Manifestation: The GitHub repository's remote URL differs between the GitHub desktop client on different operating systems (Windows/macOS) and the website terminal results. For example, on macOS, the remote URL appears as https://github.com.cnpmjs.org/youyungui/My-Knowledge-System.git, while on the GitHub desktop client on Windows and on the [www.github.com](http://www.github.com/) website, the remote URL is displayed as https://github.com/youyungui/My-Knowledge-System.git.

Methods Attempted:

(1) Tried modifying the remote URL via the GitHub desktop client, but changes did not take effect.
(2) Attempted changing it through macOS terminal by resetting the remote URL, using the command: git remote set-url origin https://github.com/youyungui/My-Knowledge-System.git. This should be a fundamental solution, but upon rechecking with git remote -v, the incorrect URL remains.
(3) Modified .git/config directly via terminal using nano .git/config, where the correct URL https://github.com/youyungui/My-Knowledge-System.git appears.
(4) Reinstalled the GitHub desktop client and cleared cache traces, but the problem persists.
Additionally, I had previously run a command to accelerate GitHub sync, which resulted in the remote URL change, and now it can't be reverted. Any help would be greatly appreciated.

**Comment by SyedAliJunaidSherazi** [comment]
It looks like you're facing issues with changing the remote URL for your GitHub repository. Here’s a step-by-step approach to resolve this:

### Steps to Fix Remote URL Issues

1. **Check Current Remote URL**: Start by verifying the current remote URL in your terminal:
   ```bash
   git remote -v
   ```
   This will list all remote URLs associated with your repository.

2. **Change Remote URL Using Terminal**: Since you've attempted this already, ensure you run the command correctly:
   ```bash
   git remote set-url origin https://github.com/youyungui/My-Knowledge-System.git
   ```
   After running this command, double-check with `git remote -v` to confirm the change.

3. **Edit `.git/config` File**: If the terminal command didn’t work, you might have already edited the `.git/config` file. Ensure that the section under `[remote "origin"]` reflects the correct URL:
   ```ini
   [remote "origin"]
       url = https://github.com/youyungui/My-Knowledge-System.git
       fetch = +refs/heads/*:refs/remotes/origin/*
   ```

4. **Use SSH Instead**: If HTTPS continues to give you trouble, consider using SSH for your repository. First, ensure you have an SSH key added to your GitHub account. Then set the remote URL like this:
   ```bash
   git remote set-url origin git@github.com:youyungui/My-Knowledge-System.git
   ```

5. **Clear Cached Credentials**: Sometimes, cached credentials may cause issues. You can clear them with:
   ```bash
   git credential-cache exit
   ```
   Then try to push or pull again to see if you're prompted for credentials.

6. **Check for Global Git Configurations**: Ensure there are no conflicting settings in your global Git configuration. You can check with:
   ```bash
   git config --global --list
   ```

7. **Reinstall Git**: If all else fails, reinstalling Git on your macOS can sometimes resolve underlying issues. Be sure to backup any configuration before doing this.

### Additional Notes

- **CNPMS**: It seems your URL is using `github.com.cnpmjs.org`, which is often a caching proxy for faster access. If you're not specifically needing that, ensure you're using the standard `github.com` URL.
- **Sync Command**: If you previously ran a command to accelerate sync, consider resetting any settings related to that.

If these steps don’t resolve your issue, please provide any error messages that appear after trying them, and I’d be happy to assist further :)

**Reply by youyungui** [reply]
Thank you so much, I'll try it soon.

**Reply by SyedAliJunaidSherazi** [reply]
No worries brother. Please don't forget to approve and accept my answer if it worked for you :)

**Reply by youyungui** [reply]
I do as follows:
1.  git remote -v
origin	https://github.com.cnpmjs.org/youyungui/My-Knowledge-System.git (fetch)
origin	https://github.com.cnpmjs.org/youyungui/My-Knowledge-System.git (push)

2.  git remote set-url origin https://github.com/youyungui/My-Knowledge-System.git

3. git remote -v
origin	https://github.com.cnpmjs.org/youyungui/My-Knowledge-System.git (fetch)
origin	https://github.com.cnpmjs.org/youyungui/My-Knowledge-System.git (push)
Things not changed.

4. nano .git/config, find its right.     Problem is not resolved.

5. git remote set-url origin git@github.com:youyungui/My-Knowledge-System.git

6. git credential-cache exit

7. git config --global --list
It comes as follow:
filter.lfs.clean=git-lfs clean -- %f
filter.lfs.smudge=git-lfs smudge -- %f
filter.lfs.process=git-lfs filter-process
filter.lfs.required=true
user.name=youyungui
user.email=youyungui@outlook.com
url.https://github.com.cnpmjs.org/.insteadof=https://github.com/

8. git config --global --unset url."https://github.com.cnpmjs.org/".insteadof 
(key step)

9. git remote -v
suprise!!!   The URL is right.
origin	git@github.com:youyungui/My-Knowledge-System.git (fetch)
origin	git@github.com:youyungui/My-Knowledge-System.git (push)

This issue is perfectly resolved.  Thank you so much!!!

**Reply by SyedAliJunaidSherazi** [reply]
Your welcome mate. am always here to help :) . please do follow my GitHub account for new updates and open source project repos.
